File: python/oop.py
# bin/usr/python3
# Create a Python class named Person.
# The Person class should have the following attributes: name, age, gender
# Implement a method called introduce that prints a message introducing the person with their name, age, and gender.
# Create an instance of the Person class and call the introduce method to display the person's information.


class Person:

    def __init__(self, name, age, gender) -> None:
        self.name = name
        self.age = age
        self.gender = gender

    def introduce(self) -> None:
        print(f"{self.name} is a {self.age} years old {self.gender}")


person = Person("John", 25, "Male")
person.introduce()

File: dart/oop.dart
// An object-oriented model that uses classes and inheritance
// A class that implements an interface
// A class that overrides an inherited method
// An instance of a class that is initialized with data from a file
// A method that demonstrates the use of a loop
void main() {
  Person person = Person('James');
  var student = Student('John', 'B', 25);

  student.display();
  person.display();
}

class Person {
  String name;

  Person(this.name);

  void display() => print('$name is a person');
}

class Student extends Person {
  String name;
  String grade;
  int age;

  Student(this.name, this.grade, this.age) : super(name);

  @override
  void display() {
    print('name: ${name}');
    print('age: $age');
    print('grade: $grade');
  }
}
